File: 83. Remove Duplicates from Sorted List/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        for (ListNode* node = head; node != nullptr; node = node->next) {
            while (node->next != nullptr && node->next->val == node->val) {
                node->next = node->next->next;
            }
        }
        return head;
    }
};

File: 83. Remove Duplicates from Sorted List/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
	for curr := head; curr != nil; curr = curr.Next {
		next := curr.Next
		for ; next != nil && next.Val == curr.Val; next = next.Next {
		}
		curr.Next = next
	}
	return head
}
